core/vm/mina_contracts.go

```diff
@@ -0,0 +1,122 @@
+package vm
+
+/*
+#cgo LDFLAGS: mina/target/release/libmina.a -ldl
+#include "../../mina/target/mina.h"
+*/
+import "C"
+import (
+	"bytes"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var sol_bool, _ = abi.NewType("bool", "", nil)
+var sol_uint8, _ = abi.NewType("uint8", "", nil)
+var sol_bytes32, _ = abi.NewType("bytes32", "", nil)
+var sol_bytes32Arr, _ = abi.NewType("bytes32[]", "", nil)
+
+type MinaPoseidon struct{}
+
+func (c *MinaPoseidon) RequiredGas(input []byte) uint64 {
+	return 1000
```

Comment:
Was there any rationale for this number? I don't think it really matters that much, but just to know